### PR TITLE
Event feedback

### LINKED ISSUE #121: Event feedback: Prompt Feedback on Attended Events
Instead of showing a list of the most recent past events on the home page which provides very little value to a user, show them a list of past events that they've attended (if they have any attendences) and ask them if they want to leave feedback for the event to help improve the quality of ACM (feedback route has existed on the portal api for years but never been used so I don't know if it works correctly and saves to a table somewhere in the database - validate this first with backend) 

- [ ] Users should be able to submit event feedback after having attended an event
- [ ] Make buttons on checkin/event modal that prompt for feedback

References:

- acmucsd/membership-portal#114
- acmucsd/membership-portal#133
- acmucsd/membership-portal#412
- [Figma: feedback submissions](https://www.figma.com/file/tv9hYPFP0HPGGXgonS5Jvw/Portal---Mobile-Only-Redesign?type=design&node-id=0-4848&mode=design&t=yEfkAYi3wJH18mmb-4)
- [Figma: feedback form](https://www.figma.com/file/tv9hYPFP0HPGGXgonS5Jvw/Portal---Mobile-Only-Redesign?type=design&node-id=0-4981&mode=design&t=yEfkAYi3wJH18mmb-4)
- [Figma: attended event feedback](https://www.figma.com/file/tv9hYPFP0HPGGXgonS5Jvw/Portal---Mobile-Only-Redesign?type=design&node-id=0-2766&mode=design&t=yEfkAYi3wJH18mmb-4)

image

image

### PR DESCRIPTION
**Pull request by SheepTester**
# Info

Closes #121 

**Please note**: Currently, the admin test account cannot submit feedback because someone requested a password reset.

Currently waiting on acmucsd/membership-portal#412, which links feedback to events

I added separate event pages and made event cards link to them. Clicking on an event card still opens the modal, but now you can ctrl + click on event cards to quickly open them in new tabs. Also, since the modal is already a form, and HTML doesn't allow forms within forms, I can't put the feedback form in a modal. So the event page is where you give feedback for events

To access the feedback form on the event page:

- Click on an event > Click on "Give feedback" (currently a barely styled link)
- Ctrl + click (or right click/long press > open in a new tab) on an event card to open the page in a new tab

To access the feedback page:

- Admin settings > Events > View Feedback
- Go directly to `/feedback` in the address bar. The API supports getting feedback for normal users, but it just lists the users' submitted feedback. This is supported by the feedback page, but I don't think it's as important to link it anywhere

I'm thinking of adding multiple-choice options to the feedback form (e.g. How did you hear about this event? What made you spend time at this event? [and if the user didn't attend the event] Why didn't you attend the event?). Since the API just stores feedback as one string, this could just be concatenated together into a both human- and program-readable string

## Changes

- [Fill in here]

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

image

image

image
image

### CODE REVIEW

File: src/components/events/EventCard/index.tsx [hunk 1]
```diff
@@ -69,16 +71,19 @@ const EventCard = ({
...
+      <Link
+        href={`${config.eventsRoute}/${uuid}`}
         data-community={community}
         className={`${styles.container} ${borderless ? '' : styles.bordered} ${className || ''}`}
-        onClick={() => setExpanded(true)}
+        onClick={e => {
+          e.preventDefault();
+          setExpanded(true);
+        }}
       >
```
Comment (alexzhang1618) [inline src/components/events/EventCard/index.tsx:82]: seeking to learn: what does this link do? Is it a fallback where there's no modal

File: src/components/events/EventDetail/index.tsx [hunk 2]
```diff
@@ -23,15 +27,31 @@ const EventDetail = ({ event, attended }: EventDetailProps) => {
...
 
+  let buttons = null;
+  if (!isOrderPickupEvent(event)) {
+    if (isUpcomingEvent) {
+      buttons = <CalendarButtons event={event} />;
+    } else if (inModal) {
+      buttons = (
+        <Link href={`${config.eventsRoute}/${event.uuid}`} className={styles.feedbackBtn}>
+          <VscFeedback aria-hidden />
+          Give feedback
```
Comment (alexzhang1618) [inline src/components/events/EventDetail/index.tsx]: Thoughts on `Add Feedback` instead of `Give Feedback`?

**Review by alexzhang1618 — APPROVED** [review]
Functionality works great! 

Some comments:
- Thoughts on adding labels for the form items? It might just be me but I started typing and didn't realize the first item was `how did you hear about this event`
Screenshot 2024-03-31 at 2 18 06 PM

File: src/components/events/EventCard/index.tsx (hunk 1 above)
Comment (SheepTester) [reply src/components/events/EventCard/index.tsx:82]: if you middle click or right click it, it doesn't call `onClick` so it behaves like a normal link, so you can open it in a new tab. if you click it normally it opens the modal

File: src/components/events/CheckInModal/index.tsx [hunk 3]
```diff
@@ -105,9 +107,14 @@ const CheckInModal = ({ open, event, onClose }: CheckInModalProps) => {
...
-          <Typography variant="h4/bold">Done</Typography>
-        </button>
+        <div className={style.buttonRow}>
+          <button type="submit" className={`${style.button} ${style.done}`}>
+            <Typography variant="h4/bold">Close</Typography>
+          </button>
+          <Link href={`${config.eventsRoute}/${event.uuid}`} className={style.button}>
+            <Typography variant="h4/bold">Add Feedback</Typography>
+          </Link>
+        </div>
```
Comment (alexzhang1618) [inline src/components/events/CheckInModal/index.tsx]: IMO add feedback before close, and maybe make add feedback the blue button and close the gray button?

Comment (alexzhang1618) [reply src/components/events/CheckInModal/index.tsx]: Also not really related to this PR but it seems like the checkin modal buttons are kinda big. I know I wrote them 😵, but can we reduce the font and padding on them a little bit? like whatever the font size for the description is is probably good

File: src/components/feedback/FeedbackForm/style.module.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    &:is(textarea) {
+      min-height: 5rem;
+      padding: 1rem;
+      resize: vertical;
+    }
+  }
+
+  .submit {
+    align-self: flex-start;
+    background-color: var(--theme-text-on-background-1);
```
Comment (alexzhang1618) [inline src/components/feedback/FeedbackForm/style.module.scss]: Can we do theme primary 2 here to be consistent?

**Review by alexzhang1618 — APPROVED** [review]
code lgtm, just some styling comments:

Can we make the background elevated for feedback here
Screenshot 2024-04-01 at 10 47 53 AM

File: src/components/events/CheckInModal/index.tsx (hunk 3 above)
Comment (SheepTester) [reply src/components/events/CheckInModal/index.tsx]: the button text is the same size as the description

**Pull request merged**